### PR TITLE
Chore: e2e tests on page level progress extension (Issue/207)

### PR DESCRIPTION
**Pull request by lemmyadams**
Addresses #207 

Adds basic level of e2e test for now, checking mainly if the bar appears when it should

### CODE REVIEW

File: test/e2e/pageLevelProgress.cy.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+describe('Page Level Progress', function () {
+  beforeEach(function () {
+    cy.getData()
+    cy.visit('/')
+  });
+  
+  it('should display the page level progress bars correctly on the menu items', function () {
+    const isPageLevelProgressEnabled = this.data.course._pageLevelProgress._isEnabled
```
Comment (cahirodoherty-learningpool) [inline test/e2e/pageLevelProgress.cy.js:8]: We might need to guard here for courses that don't have the extension added
```suggestion
    const isPageLevelProgressEnabled = this.data.course._pageLevelProgress?._isEnabled
```

File: test/e2e/pageLevelProgress.cy.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    // Check if PLP is enabled. Check it's visible on menu tiles
+    if (isPageLevelProgressEnabled) {
+      const pagesCount = this.data.contentObjects.filter((page) => page._pageLevelProgress._isEnabled === true).length
```
Comment (cahirodoherty-learningpool) [inline test/e2e/pageLevelProgress.cy.js:11]: The _isEnabled value is boolean, so we don't need triple equals here
```suggestion
      const pagesCount = this.data.contentObjects.filter((page) => page._pageLevelProgress._isEnabled).length
```

File: test/e2e/pageLevelProgress.cy.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  });
+  
+  it('should display the page level progress bars correctly on the pages', function () {
+    const pageLevelProgress = this.data.course._pageLevelProgress
+    if (!pageLevelProgress._isEnabled) return
+    const pages = this.data.contentObjects
+    pages.forEach((page) => {
+      cy.visit(`/#/${page._id}`);
+      // Only check it appears correctly if it shows in the nav bar and its enabled on the page
+      if (page._pageLevelProgress._isEnabled === true && pageLevelProgress._isShownInNavigationBar) {
```
Comment (cahirodoherty-learningpool) [inline test/e2e/pageLevelProgress.cy.js:25]:
```suggestion
      if (page._pageLevelProgress._isEnabled && pageLevelProgress._isShownInNavigationBar) {
```

File: test/e2e/pageLevelProgress.cy.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      const pagesCount = this.data.contentObjects.filter((page) => page._pageLevelProgress._isEnabled).length
+      cy.get('.pagelevelprogress__indicator').should('have.length', pagesCount)
+    } else {
+      cy.get('.pagelevelprogress__indicator').should('not.exist')
+    }
+  });
+  
+  it('should display the page level progress bars correctly on the pages', function () {
+    const pageLevelProgress = this.data.course._pageLevelProgress
+    if (!pageLevelProgress._isEnabled) return
```
Comment (cahirodoherty-learningpool) [inline test/e2e/pageLevelProgress.cy.js:20]:
```suggestion
    if (!pageLevelProgress?._isEnabled) return
```

File: test/e2e/pageLevelProgress.cy.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        const plpComponents = componentsOnPage.filter((component) => component._pageLevelProgress?._isEnabled)
+        cy.get('.pagelevelprogress__indicator').should('exist')
+        cy.get('button.nav__pagelevelprogress-btn').click()
+        // TODO: If its a random assessment more checks are necessary
+        if (page._classes !== 'assessment') {
+          cy.get('.pagelevelprogress__item').should('have.length', plpComponents.length)
+        }
+      } else {
+        cy.get('.pagelevelprogress__indicator').should('not.exist')
+      }
```
Comment (cahirodoherty-learningpool) [inline test/e2e/pageLevelProgress.cy.js:38]: Maybe flip this for easier reading?
```suggestion
      if (!page._pageLevelProgress?._isEnabled || pageLevelProgress._isShownInNavigationBar) {
        cy.get('.pagelevelprogress__indicator').should('not.exist')
        return
      }
      
      const articlesOnPage = this.data.articles.filter((article) => article._parentId === page._id).map(article => article._id)
      const blocksOnPage = this.data.blocks.filter((block) => articlesOnPage.includes(block._parentId)).map(blocks => blocks._id)
      const componentsOnPage = this.data.components.filter((component) => blocksOnPage.includes(component._parentId))
      const plpComponents = componentsOnPage.filter((component) => component._pageLevelProgress?._isEnabled)
      cy.get('.pagelevelprogress__indicator').should('exist')
      cy.get('button.nav__pagelevelprogress-btn').click()
      // TODO: If its a random assessment more checks are necessary
      if (page._classes !== 'assessment') {
        cy.get('.pagelevelprogress__item').should('have.length', plpComponents.length)
      }
```

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: test/e2e/pageLevelProgress.cy.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    } else {
+      cy.get('.pagelevelprogress__indicator').should('not.exist')
+    }
+  });
+  
+  it('should display the page level progress bars correctly on the pages', function () {
+    const pageLevelProgress = this.data.course._pageLevelProgress
+    if (!pageLevelProgress?._isEnabled) return
+    const pages = this.data.contentObjects
+    pages.forEach((page) => {
```
Comment (joe-allen-89) [inline test/e2e/pageLevelProgress.cy.js:22]:
```suggestion
    pages.forEach(page => {
```

**Review by joe-allen-89 — APPROVED** [review]
👍

**Pull request merged**